scripts/kind-e2e/lib_subctl_gather_test.sh

```diff
@@ -11,7 +11,7 @@ function test_subctl_gather() {
   rm -rf $out_dir
   mkdir $out_dir
 
-  ${DAPPER_SOURCE}/bin/subctl gather --kubeconfig ${KUBECONFIGS_DIR}/kind-config-${cluster} --dir $out_dir
+  ${DAPPER_SOURCE}/bin/subctl gather --dir $out_dir
```

Comment:
Since we're now gathering from all clusters, we should verify each one. The validate functions below reference `${cluster}` which was being set by `with_context`. I suggest we put L16-47 below in a `validate_gathered_files` function and invoke via:

```
for cluster in "${clusters[@]}"; do
    with_context "${cluster}" validate_gathered_files
done
```

We could use `run_all_clusters` but that will do in parallel which will make the output harder to follow.